### ISSUE TITLE
[BUG/ISSUE] Nested run overflows at hcox_paranox_mod.F90

### ISSUE THREAD

**Issue by xpye98** [geoschem/geos-chem#526]
# Report a GEOS-Chem bug or technical issue

## Describe the bug:
Nested run crashed at the beginning of time step when HEMCO enters `Evolve_Plume` function in `hcox_paranox_mod.F90`

### Expected behavior: ###

Run nested simulation

### Steps to reproduce: the bug:

I ran the global 2x2.5 simulation and output BC data (I set the range of BC exactly the same as my nested run region) successfully. Then, I used restart file from 2x2.5 simulation to run nested simulation in Asia. 
It looks fine all the way until the beginning of time step. As I can see from `HEMCO.log` file, the HEMCO entered `Evolve_Plume` function in `hcox_paranox_mod.F90`, and it then overflows. The debug output says `forrtl: error (72): floating overflow` and it happens at line 2557 in ` hcox_paranox_mod.F90`.

I went back to the line 2557, which is
```
 VARS(3) = ExtState%O3%Arr%Val(I,J,1)         &
           * HcoState%Phys%AIRMW        / Inst%MW_O3 &
           * 1.e9_sp
```
So, I guess maybe O3 concentration here is out of range. Then I just printed [O3] out and found it's  `-1.656886409097943E+031`, which is of course not right.

I don't know where goes wrong, and hopes you can get me through it.

### Error messages

```
forrtl: error (72): floating overflow
Image              PC                Routine            Line        Source
geos               0000000001DC87F9  Unknown               Unknown  Unknown
geos               0000000001DC70CE  Unknown               Unknown  Unknown
geos               0000000001D776A2  Unknown               Unknown  Unknown
geos               0000000001D15938  Unknown               Unknown  Unknown
geos               0000000001D1D0CB  Unknown               Unknown  Unknown
libpthread.so.0    00002ABB2B271370  Unknown               Unknown  Unknown
geos               00000000012D2255  hcox_paranox_mod_        2564  hcox_paranox_mod.F90
geos               0000000001297C0B  hcox_paranox_mod_         538  hcox_paranox_mod.F90
geos               0000000001294AF6  hcox_paranox_mod_         314  hcox_paranox_mod.F90
geos               00000000010A5540  hcox_driver_mod_m         550  hcox_driver_mod.F90
geos               0000000000702CCA  hcoi_gc_main_mod_         934  hcoi_gc_main_mod.F90
geos               00000000005AD78C  emissions_mod_mp_         202  emissions_mod.F90
geos               000000000040FA83  MAIN__                   1494  main.F90
geos               000000000040AD96  Unknown               Unknown  Unknown
libc.so.6          00002ABB2B6A3B35  Unknown               Unknown  Unknown
geos               000000000040AC79  Unknown               Unknown  Unknown
```

### Your GEOS-Chem version and runtime environment:

 - GEOS-Chem version: 12.9.3
 - Compiler version: Intel 14.0.2.20140120
 - Did you run on a computational cluster, on the AWS cloud: no
 - Are you using GEOS-Chem "out of the box" (i.e. unmodified): 
   - If you have modified GEOS-Chem, please list what was changed:  I changed some emission file but it should not be related to this error

### Input and log files to attach

[slurm_out.txt](https://github.com/geoschem/geos-chem/files/5559486/slurm_out.txt)
[HEMCO_log.txt](https://github.com/geoschem/geos-chem/files/5559488/HEMCO_log.txt)
[HEMCO_Config.txt](https://github.com/geoschem/geos-chem/files/5559489/HEMCO_Config.txt)
[input.txt](https://github.com/geoschem/geos-chem/files/5559490/input.txt)
[log.txt](https://github.com/geoschem/geos-chem/files/5559491/log.txt)

However, I tested another simulation time, it works with no errors.

**Comment by yantosca** [comment]
Thanks for writing @xpye98.  @MaggieMarvin reported this similar issue a while back, and also gave a fix for it  Please see issue #243.  

I think this issue is caused by compiler optimization and/or roundoff that results in an unphysical value for solar elevation angle.  Applying the fix in #243 should get your run past that problem.

**Comment by xpye98** [comment]
Thanks @yantosca , it somehow works now.
